sys/net/gnrc/conn/tcp/gnrc_conn_tcp.c

```diff
@@ -0,0 +1,560 @@
+{
+    assert(conn->l4_type == GNRC_NETTYPE_TCP);
+    memcpy(addr, &(conn->peer_addr), conn->peer_addr_len);
+    *port = conn->peer_port;
+    return conn->peer_addr_len;
+}
+
+int conn_tcp_connect(conn_tcp_t *conn, const void *addr, size_t addr_len, uint16_t port)
+{
+    msg_t msg;
+    msg_t msg_tout;
+    xtimer_t usr_tout;
+    int err = 0;
+
+    assert(conn->l4_type == GNRC_NETTYPE_TCP);
+
+    /* Connection is already connected: Return -EISCONN */
+    if (conn->state != CLOSED) {
+        return -EISCONN;
+    }
+
+    /* This call is only allowed on active Connections*/
+    if (conn->status & GNRC_TCP_STATUS_PASSIVE) {
+        return -EOPNOTSUPP;
+    }
+
+    /* Validate IP-Address format */
+    switch (conn->l3_type) {
+#ifdef MODULE_GNRC_IPV6
+        case GNRC_NETTYPE_IPV6:
+            if (addr != NULL && addr_len == sizeof(ipv6_addr_t)) {
+                if (ipv6_addr_is_unspecified((ipv6_addr_t *) addr)) {
+                    return -EDESTADDRREQ;
+                }
+                break;
+            }
+#endif
+        default:
+            return -EAFNOSUPPORT;
+    }
+
+    /* Copy Peer information into conn_tcp_t structure */
+    memcpy(conn->peer_addr, addr, addr_len);
+    conn->peer_addr_len = addr_len;
+    conn->peer_port = port;
+    conn->status = 0;
+
+    /* Reset RTO values */
+    conn->srtt = GNRC_TCP_RTO_UNINITIALIZED;
+    conn->rtt_var = GNRC_TCP_RTO_UNINITIALIZED;
+    conn->rto = GNRC_TCP_RTO_UNINITIALIZED;
+
+    /* Take ownership for this connection, attach message queue */
+    conn->owner = thread_getpid();
+    msg_init_queue(conn->msg_queue, GNRC_TCP_MSG_QUEUE_SIZE);
+
+    /* Purge message queue */
+    while(msg_try_receive(&msg) >= 0){
+    }
+
+    /* Setup User Space Timeout */
+    msg_tout.type = MSG_TYPE_USER_TIMEOUT;
+    xtimer_set_msg(&usr_tout, GNRC_TCP_USER_TIMEOUT_DURATION, &msg_tout, conn->owner);
+
+    /* Call FSM with Event: CALL_OPEN */
+    if (_fsm(conn, CALL_OPEN, NULL, NULL, 0) == -ENOMEM) {
+        DEBUG("gnrc_conn_tcp : conn_tcp_connect() : Out of receive buffers, increase GNRC_TCP_RCV_BUFFERS\n");
+        err = -ENOMEM;
+    }
+
+    /* Wait until a connection was established or closed */
+    while (err >= 0 && conn->state != CLOSED && conn->state != ESTABLISHED && conn->state != CLOSE_WAIT) {
+        msg_receive(&msg);
+        switch (msg.type) {
+            case MSG_TYPE_USER_TIMEOUT:
+                DEBUG("gnrc_conn_tcp : conn_tcp_connect() : USER_TIMEOUT\n");
+                _fsm(conn, TIMEOUT_USER, NULL, NULL, 0);
+                err = -ETIME;
+                break;
+
+            case MSG_TYPE_NOTIFY_USER:
+                DEBUG("gnrc_conn_tcp : conn_tcp_connect() : MSG_TYPE_NOTIFY_USER\n");
+                break;
+
+            default:
+                DEBUG("gnrc_conn_tcp : conn_tcp_open() : other message type\n");
+        }
+    }
+
+    /* Cleanup */
+    if(err != -ETIME){
+      xtimer_remove(&usr_tout);
```

Comment:
Won't fix. Depricated, conn implementation will be removed after sock_tcp release.